### PR TITLE
Add readthedoc configuration

### PR DESCRIPTION
**Pull request by RemiLehe**
This adds the openPMD-viewer documentation in readthedocs:
[openpmd-viewer.readthedocs.io/](https://openpmd-viewer.readthedocs.io/)
(Right now ReadtheDoc builds from the `doc` branch ; but once this PR is merged, I will change this to `dev`.)

### CODE REVIEW

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-20.04
+  tools:
+    python: "3.11"
+
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+    configuration: doc/source/conf.py
```
Comment (RemiLehe) [inline .readthedocs.yaml]:
```suggestion
    configuration: docs/source/conf.py
```

File: docs/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sphinx
+sphinx_rtd_theme
```
Comment (RemiLehe) [inline docs/requirements.txt:3]:
```suggestion
sphinx_rtd_theme
nbsphinx
```

File: .readthedocs.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
+version: 2
+
+# Set the version of Python and other tools you might need
+build:
+  os: ubuntu-20.04
```
Comment (ax3l) [inline .readthedocs.yaml]: Want to go to 22.04? We just fixed some build issues on 20.04 the other week for all of RTD/Sphinx:
https://github.com/ECP-WarpX/WarpX/blob/23.06/.readthedocs.yml
```suggestion
  os: ubuntu-22.04
```

File: docs/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
```
Comment (ax3l) [inline docs/requirements.txt:3]: Let's add this, allows us to make tabs, etc. - super useful :)
```suggestion
sphinx-design
sphinx_rtd_theme
```
https://github.com/ECP-WarpX/WarpX/pull/3361

File: docs/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+jupyter
+nbsphinx
+wget
+numba
+numpy
+matplotlib
+scipy
```
Comment (ax3l) [inline docs/requirements.txt:11]: For code highlighting in blocks:
```suggestion
pygments
scipy
```

**Review by ax3l — APPROVED** [review]

**Pull request merged**